src/Microsoft.DotNet.CodeFormatter.Analyzers.Tests/ExplicitVariableTypeAnalyzerTests.cs

```diff
@@ -0,0 +1,408 @@
+class C1
+{
+    bool[] T() 
+    {
+        return new[] { true };
+    }
+
+    void M(int a, bool[] b)
+    {
+        var x = a;
+        var y = T();  
+        var z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };
+    }
+}";
+            const string expected = @"
+class C1
+{
+    bool[] T() 
+    {
+        return new[] { true };
+    }
+
+    void M(int a, bool[] b)
+    {
+        int x = a;
+        bool[] y = T();
+        int[][] z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };
+    }
+}";
+            Verify(text, expected, runFormatter: false);
+        }
+
+        [Fact]
+        public void TestUserDefinedTypeExplicitDeclaration()
+        {
+            const string text = @"
+class C1
+{
+    class U1
+    {}
+
+    struct U2
+    {}
+
+    enum U3 { E1, E2 }
+
+    void M()
+    {
+        U1 x = new U1();
+        U2 y = new U2();
+        U3 z = U3.E1;        
+    }
+}";
+            Verify(text, text, runFormatter: false);
+        }
+
+        [Fact]
+        public void TestUserDefinedObviousTypeVarDeclaration()
+        {
+            const string text = @"
+class C1
+{
+    class U1
+    {}
+
+    struct U2
+    {}
+
+    enum U3 { E1, E2 }
+
+    void M()
+    {
+        var x = new U1();
+        var y = new U2();
+        var z = U3.E1;         
+    }
+}";
+            const string expected = @"
+class C1
+{
+    class U1
+    {}
+
+    struct U2
+    {}
+
+    enum U3 { E1, E2 }
+
+    void M()
+    {
+        var x = new U1();
+        var y = new U2();
+        U3 z = U3.E1;         
+    }
+}";
+            Verify(text, expected, runFormatter: false);
+        }
+
+        [Fact]
+        public void TestUserDefinedNonobviousTypeVarDeclaration()
+        {
+            const string text = @"
+class C1
+{
+    class U1
+    {}
+
+    struct U2
+    {}
+
+    enum U3 { E1, E2 }
+
+    void M(U1 u1, U2 u2, U3 u3)
+    {
+        var x = u1;
+        var y = u2;
+        var z = u3;         
+    }
+}";
+            const string expected = @"
+class C1
+{
+    class U1
+    {}
+
+    struct U2
+    {}
+
+    enum U3 { E1, E2 }
+
+    void M(U1 u1, U2 u2, U3 u3)
+    {
+        U1 x = u1;
+        U2 y = u2;
+        U3 z = u3;         
+    }
+}";
+            Verify(text, expected, runFormatter: false);
+        }
+
+        [Fact]
+        public void TestSimpleForEachExplicitDeclaration()
+        {
+            const string text = @"
+class C1
+{
+    void M(string[] a1, string[][] a2)
+    {
+        foreach (string element in a1)
+        { }  
+ 
+        foreach (string[] element in a2)
+        { }
+    }
+}";
+            Verify(text, text, runFormatter: false);
+        }
+
+        [Fact]
+        public void TestSimpleForEachObviousVarDeclaration()
+        {
+            const string text = @"
+class C1
+{
+    void M()
+    {
+        foreach (var element in new string[] { "" })
+        { }  
+ 
+        foreach (var element in new int[] { new int[] { 1, 2, 3 }, new int[] { 4, 5, 6 } })
```

Comment:
@lgolding In the test you mentioned, it is `var z = new[] { new[] { 1, 2, 3 }, new[] { 4, 5, 6 } };`, which is an anonymous array creation such that the type is not given directly in the declaration like a regular array declaration (even though it is kinda obvious in this case).